### PR TITLE
Close #180 - Support Dotty (Scala 3.0.0-M1)

### LINKED ISSUE #180: Support Dotty (Scala 3.0.0-M1)
Since Scala 3.0.0-M1 (Dotty) has been released, I think it's time to support Dotty.
I'm currently working on it but having an issue with macro used in Minitest.
Fortunately, there is an open [PR](https://github.com/monix/minitest/pull/66) to support Dotty on the Minitest repo so I'll finish it once that PR is merged and released.

### PR DESCRIPTION
**Pull request by Kevin-Lee**
Close #180 - Support Dotty (Scala 3.0.0-M1)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -31,8 +39,8 @@ ThisBuild / scmInfo := Some(
       "scm:git@github.com:hedgehogqa/scala-hedgehog.git"
     )
   )
-ThisBuild / scalaVersion := "2.12.11"
-ThisBuild / crossScalaVersions := Seq("2.11.12", scalaVersion.value, "2.13.3")
+ThisBuild / scalaVersion := "2.13.3"
+ThisBuild / crossScalaVersions := Seq("2.11.12", "2.12.12", "2.13.3", "3.0.0-M1")
```
Comment (charleso) [inline build.sbt:43]: Should probably make 2.13.3 here `scalaVersion.value`?

File: build.sbt [hunk 2]
```diff
@@ -111,35 +135,62 @@ lazy val testJS = test.js
...
+        }
+      )
+    })
   , scalacOptions in (Compile,console) := Seq("-language:_", "-feature")
   , scalacOptions in (Test,console) := Seq("-language:_", "-feature")
   , unmanagedSourceDirectories in Compile ++= {
       (unmanagedSourceDirectories in Compile).value.map { dir =>
         CrossVersion.partialVersion(scalaVersion.value) match {
-          case Some((2, 13)) => file(dir.getPath ++ "-2.13+")
+          case Some((2, 13)) | Some((3, 0)) => file(dir.getPath ++ "-2.13+")
```
Comment (charleso) [inline build.sbt:181]: Should we match just on 3.x here to make it future compatible? Or not bother?

**Review by charleso — APPROVED** [review]
Nice

File: build.sbt (hunk 1 above)
Comment (Kevin-Lee) [reply build.sbt:43]: Ah! Yeah I will fix it. Thanks!

File: build.sbt (hunk 2 above)
Comment (Kevin-Lee) [reply build.sbt:181]: As far as I know Scala 2.13 source code is compatible with Scala 3.0 so it should be fine. If it's just `scala-3.0`, I think, sbt can handle it.

**Pull request merged**